hamilton/cli/__main__.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from pathlib import Path
+from pprint import pprint
+from typing import Annotated, Any, Optional
+
+import typer
+from pydantic import BaseModel
+
+from hamilton import driver
+from hamilton.cli import commands
+
+logger = logging.getLogger(__name__)
+
+
+class Response(BaseModel):
+    command: str
+    success: bool
+    message: Any
+
+
+class CliState:
+    verbose: Optional[bool] = None
+    json_out: Optional[bool] = (None,)
+    dr: Optional[driver.Driver] = None
+    dataflow_version: Optional[dict] = None
+
+
+cli = typer.Typer()
+state = CliState()
+
+
+# entrypoint for `hamilton` without command
+@cli.callback()
+def main(
+    verbose: Annotated[bool, typer.Option(help="Output all intermediary commands")] = False,
+    json_out: Annotated[
+        bool, typer.Option(help="Output JSON for programmatic use (e.g., CI)")
+    ] = False,
+):
+    """Hamilton CLI"""
+    state.verbose = verbose
+    state.json_out = json_out
+
+
+@cli.command()
+def build(
+    ctx: typer.Context,
+    modules: Annotated[
+        list[Path], typer.Argument(exists=True, dir_okay=False, readable=True, resolve_path=True)
+    ],
+):
+    """Build dataflow with MODULES"""
+    try:
+        config = dict()
+        state.dr = commands.build(modules=modules, config=config)
+    except Exception as e:
+        response = Response(
+            command="build", success=False, message={"error": str(type(e)), "details": str(e)}
+        )
+        print(response.model_dump_json())
+        raise typer.Exit(code=1)
+
+    response = Response(
+        command="build", success=True, message={"modules": [p.stem for p in modules]}
+    )
+
+    if (ctx.info_name == "build") or state.verbose:
```

Comment:
You should consider using `pprint` -- it'll format json nicely.

Reply:
I was afraid that pretty printing the JSON output would create a multiline mess for those trying to capture the `stdout`. In verbose mode, you get one JSON object per line, one for each intermediary command executed